### PR TITLE
Include /etc/hosts coredns mounts for dns=none clusters

### PR DESCRIPTION
**Pull request by rifelpet**
Ref: https://github.com/kubernetes/kops/pull/16335#issuecomment-1937413388

### CODE REVIEW

**Comment by rifelpet** [comment]
cc @hakman 

You're right, this is more relevant to the failing tests. I'm not sure what the headless services in https://github.com/kubernetes/kops/pull/16335 are actually used for. Do you happen to know?

File: upup/models/cloudup/resources/addons/coredns.addons.k8s.io/k8s-1.12.yaml.template [hunk 1]
```diff
@@ -76,7 +76,7 @@ data:
           fallthrough in-addr.arpa ip6.arpa
           ttl 30
         }
-    {{- if GossipName }}
+    {{- if not PublishesDNSRecords }}
         hosts /rootfs/etc/hosts k8s.local {
```
Comment (rifelpet) [inline upup/models/cloudup/resources/addons/coredns.addons.k8s.io/k8s-1.12.yaml.template:80]: /hold Actually this needs to be updated to the cluster's domain

Comment (rifelpet) [reply upup/models/cloudup/resources/addons/coredns.addons.k8s.io/k8s-1.12.yaml.template]: /hold cancel

**Review by hakman — APPROVED** [review]

**Comment by rifelpet** [comment]
Confirmed this fixed the e2e failures: https://testgrid.k8s.io/kops-grid#kops-grid-calico-flatcar-k28

**Pull request merged**